subprojects/platform-native/src/main/java/org/gradle/nativeplatform/tasks/LinkSharedLibrary.java

```diff
@@ -39,6 +42,16 @@ public void setInstallName(String installName) {
         this.installName = installName;
     }
 
+    /**
+     * The file where the linked binary will be located.
+     *
+     * @since 4.4
+     */
+    @Internal
+    public RegularFileProperty getLinkFile() {
```

Comment:
If we add this to the task, then the task should honour this setting by driving the linker to make sure the file ends up in the specified location (and it should be an `@OutputFile`).

It belongs here, but not as a partial implementation. We should either add it later as a separate PR or finish up the implementation in this PR. I'd probably just leave it out for now.